contracts/sol6/KyberNetwork.sol

```diff
@@ -579,71 +575,61 @@ contract KyberNetwork is WithdrawableNoModifiers, Utils5, IKyberNetwork, Reentra
         return;
     }
 
-    /// @dev call trade from reserve and verify balances
-    ///      return actual dest amount received from reserve
-    /// @param reserve reserve address to trade with
+    /// @dev call trade from reserves and verify balances
+    /// @param reservesData reservesData to trade
     /// @param src src token of the trade
-    /// @param srcAmount amount of src token to trade
     /// @param dest dest token of the trade
-    /// @param conversionRate conversion rate of the trade
-    /// @param srcDecimals src token decimals
-    /// @param destDecimals dest token decimals
+    /// @param srcDecimals decimals of src token
+    /// @param destDecimals decimals of dest token
     function tradeAndVerifyNetworkBalance(
-        IKyberReserve reserve,
+        ReservesData memory reservesData,
         IERC20 src,
-        uint256 srcAmount,
         IERC20 dest,
-        uint256 conversionRate,
         uint256 srcDecimals,
         uint256 destDecimals
     ) internal
     {
+        // only need to verify src balance if src is not ETH
+        uint256 srcBalanceBefore = (src == ETH_TOKEN_ADDRESS) ? 0 : getBalance(src, address(this));
```

Comment:
can remove (src == ETH_TOKEN_ADDRESS)
`uint256 srcBalanceBefore = getBalance(src, address(this));`

Reply:
if src token is ETH, we don't need to verify src token balance as we will pass this value to reserve's trade function

Reply:
Yes, so it should be the same as the line below right?
`uint256 srcBalanceBefore = (src == ETH_TOKEN_ADDRESS) ? 0 : getBalance(src, address(this));`
should be
`uint256 srcBalanceBefore = getBalance(src, address(this));`
like the line below?

Reply:
if we don't need to verify src token balance, why do we need to get network's balance of src token?

Reply:
Ahhhhhhhh righhtttt, ok got it